core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
> No. I was always talking about the embedded and smart connector. Not about jdbc versus scala code.

So you are saying that in smart connector mode when writing SQL code (e.g. via SparkSession.sql), users will continue using SparkContext API, but when writing the same code in jobs they will use the new property? Why would that be the case or why would we recommend something like that?

The only valid case looks to be that users will use the SparkContext way when writing code, while they use the new property for SQL.

> SparkContext.setLocalProperty API creates a new pool if it does not already exist. We cannot allow that.

That doesn't make sense. How can we prevent users from creating new pools using SparkContext API? We will, of course, have some pools in the product default configuration but users can always create more or change those.